### ISSUE TITLE
In Slick3, slick.jdbc.JdbcBackend is not a legal prefix for a constructor

### ISSUE THREAD

**Issue by j5ik2o** [wvlet/airframe#249]
Could you solve this problem?

```scala
val dbConfig: DatabaseConfig[JdbcProfile]  = DatabaseConfig.forConfig[JdbcProfile](path = "jdbc")

val design = newDesign
          .bind[JdbcProfile].toInstance(dbConfig.profile)
          .bind[JdbcProfile#Backend#Database].toInstance(dbConfig.db)
```

error detail:

```
Error:(25, 16) slick.jdbc.JdbcBackend is not a legal prefix for a constructor
          .bind[JdbcProfile#Backend#Database].toInstance(dbConfig.db)
```

**Comment by xerial** [comment]
Airframe tires to generate a factory `new JdbcProfile#Backend#Database{}`, but this code throws the above error. The same error occurs when trying to instantiate path dependent types directly:
https://stackoverflow.com/questions/5976791/is-a-path-dependent-type-a-subtype 

Airframe needs to detect if a given type is a path dependent types or not.

**Comment by xerial** [comment]
@j5ik2o The fix is available in 0.65+2-d2433bbd-SNAPSHOT. Please try this version.
